outputs.tf

```diff
@@ -54,8 +54,8 @@ output "http_application_routing_zone_name" {
   value = azurerm_kubernetes_cluster.main.http_application_routing_zone_name != null ? azurerm_kubernetes_cluster.main.http_application_routing_zone_name : ""
 }
 
-output "system_assigned_identity" {
-  value = azurerm_kubernetes_cluster.main.identity
+output "cluster_identity" {
+  value = try(azurerm_kubernetes_cluster.main.identity[0], null)
```

Comment:
Why the `try` block ? Is there a condition when the cluster has `null` identity ?

Reply:
Why we output in the value only the first element, rather than making available the all list ?

If we export the all list the user of the module can use the output with a index and consume all the identities of the cluster in other resources.

This would be also consistent with how we export the kubelet identity, example:
https://github.com/zioproto/istio-aks-example/blob/main/registry.tf#L10

Reply:
Hi, according to the [resource schema](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/internal/services/containers/kubernetes_cluster_resource.go#L238) and [identity schema it referenced](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/vendor/github.com/hashicorp/go-azure-helpers/resourcemanager/commonschema/identity_system_or_user.go#L19-L24), the identity in aks resource is a list with maximum length 1. So in this case the identity list has three cases: `null`, an empty list and a list with exactly one item.

Reply:
What happens when you use `SystemAssigned, UserAssigned` ? Dont you have 2 cluster identities ?

Reply:
Should we also patch the [CHANGLOG.md](https://github.com/Azure/terraform-azurerm-aks/blob/master/CHANGLOG.md) to indicate that the output name changed, so folks using the module must update their code ?